dfusion_rust_core/src/models/solution.rs

```diff
@@ -3,24 +3,30 @@ use super::*;
 use log::info;
 
 use byteorder::{BigEndian, ByteOrder};
+use std::collections::HashMap;
 use std::iter::once;
 
 #[derive(Clone, Debug, PartialEq)]
 pub struct Solution {
-    pub prices: Vec<u128>,
+    // token_id => price
```

Comment:
nit: I would use `///` so this adds a doc-comment instead of just a regular comment here as this documentation for the struct field.

Reply:
I have tried this in the past and gotten strange compile errors, but will try again.